### ISSUE TITLE
Memberlist: Clicking Forget on ring pages sometimes sometimes still shows the instance

### ISSUE THREAD

**Issue by stevesg** [grafana/mimir#3340]
#### Describe the bug
When clicking Forget on the ring pages (e.g. forgetting ingesters via e.g. the distributor ring), when the page refreshes the instance is still there. I think this is possibly just a race condition between the instance being deleted from the ring, and the state used by the web interface to render the page has not got the updated state yet.

As a side effect, if one needs to forget many instances, then one will often mistakenly click the same button again and this reveals a cryptic message:

![Screenshot from 2022-10-27 12-32-47](https://user-images.githubusercontent.com/78375245/198970676-e53170cf-fe84-4399-a9c9-e39c84a59461.png)

Perhaps we can also improve this message (e.g. "Ring member not found").

#### To Reproduce

1. Start Mimir (SHA or version)
2. Start some ingesters
3. Navigate to a ring admin page on e.g. distributors
4. Click Forget on

#### Expected behavior

The instance should not be visible on the list when the page is loaded.

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: jsonnet

**Comment by pstibrany** [comment]
> I think this is possibly just a race condition between the instance being deleted from the ring, and the state used by the web interface to render the page has not got the updated state yet.

Yes, it is. This is complicated by the fact that frontend proxy will typically send each request to a ring page to different instance (say another distributor), and "forget" may not have yet propagated to all replicas at that point.

I mentioned this problem in the blog post about gossiping ring as well: https://grafana.com/blog/2020/03/25/how-were-using-gossip-to-improve-cortex-and-loki-availability/ (What did we get by doing this?)

**Comment by stevesg** [comment]
I see that pages served by _other_ instances, would not be up to date, of course, but when the page is served by instance that initiated the forget, it should know immediately that instance is gone?

**Comment by pstibrany** [comment]
> I see that pages served by _other_ instances, would not be up to date, of course, but when the page is served by instance that initiated the forget, it should know immediately that instance is gone?

Yes, that instance should would show correct result, because "forget" (ie. deletion from KV store) has already been applied to local copy of the ring.
